core/dragged_connection_manager.js

```diff
@@ -0,0 +1,208 @@
+  /**
+   * The top block in the stack being dragged.
+   * Does not change during a drag.
+   * @type {!Blockly.Block}
+   * @private
+   */
+  this.topBlock_ = block;
+
+  /**
+   * The workspace on which these connections are being dragged.
+   * Does not change during a drag.
+   * @type {!Blockly.WorkspaceSvg}
+   * @private
+   */
+  this.workspace_ = block.workspace;
+
+  /**
+   * The connections on the dragging blocks that are available to connect to
+   * other blocks.  This includes all open connections on the top block, as well
+   * as the last connection on the block stack.
+   * Does not change during a drag.
+   * @type {!Array.<!Blockly.RenderedConnection>}
+   * @private
+   */
+  this.availableConnections_ = [];
+
+  /**
+   * The connection that this block would connect to if released immediately.
+   * Updated on every mouse move.
+   * @type {Blockly.RenderedConnection}
+   * @private
+   */
+  this.closestConnection_ = null;
+
+  /**
+   * The connection that would connect to this.closestConnection_ if this block
+   * were released immediately.
+   * Updated on every mouse move.
+   * @type {Blockly.RenderedConnection}
+   * @private
+   */
+  this.localConnection_ = null;
+
+  /**
+   * The distance between this.closestConnection_ and this.localConnection_.
+   * Updated on every mouse move.
+   * @type {number}
+   * @private
+   */
+  this.radiusConnection_ = 0;
+
+  this.setAvailableConnections_();
+  Blockly.selected = block;
+};
+
+/**
+ * Update highlighted connections and the cursor based on the most recent move
+ * event.
+ * @param {!Event} e Mouse move event to respond to.
+ * @param {!goog.math.Coordinate} dxy Position relative to drag start.
+ */
+Blockly.DraggedConnectionManager.prototype.update = function(e, dxy) {
+  var oldClosestConnection = this.closestConnection_;
+  var closestConnectionChanged = this.updateClosest_(dxy);
+
+  if (closestConnectionChanged) {
+    if (oldClosestConnection) {
+      oldClosestConnection.unhighlight();
+    }
+  }
+
+  var wouldDeleteBlock = this.updateCursor_(e);
+
+  if (!wouldDeleteBlock && closestConnectionChanged &&
+      this.closestConnection_) {
+    this.addConnectionHighlighting();
+  }
+};
+
+/**
+ * Remove highlighting from the currently highlighted connection, if it exists.
+ */
+Blockly.DraggedConnectionManager.prototype.removeConnectionHighlighting = function() {
+  if (this.closestConnection_) {
+    this.closestConnection_.unhighlight();
+  }
+};
+
+/**
+ * Add highlighting to the closest connection, if it exists.
+ */
+Blockly.DraggedConnectionManager.prototype.addConnectionHighlighting = function() {
+  if (this.closestConnection_) {
+    this.closestConnection_.highlight();
+  }
+};
+
+/**
+ * Populate the list of available connections on this block stack.  This should
+ * only be called once, at the beginning of a drag.
+ * @private
+ */
+Blockly.DraggedConnectionManager.prototype.setAvailableConnections_ = function() {
```

Comment:
This used to be being recalculated on every mouse move, which is a little bit ridiculous.